src/OrleansRuntime/GrainTypeManager/GrainTypeManager.cs

```diff
@@ -143,11 +145,36 @@ internal void SetInterfaceMapsBySilo(Dictionary<SiloAddress, GrainInterfaceMap>
             RebuildFullGrainInterfaceMap();
         }
 
-        internal IList<SiloAddress> GetSupportedSilos(int typeCode)
+        internal IReadOnlyList<SiloAddress> GetSupportedSilos(int typeCode)
         {
             return supportedSilosByTypeCode[typeCode];
         }
 
+        internal IReadOnlyList<SiloAddress> GetSupportedSilos(int typeCode, int ifaceId, IList<ushort> versions)
+        {
+            var result = new List<SiloAddress>();
+
```

Comment:
Yes we can, but it is a bit hard to cache since we are taking a list of versions as an input. I think I can refactor to take only one and add caching.